### PR TITLE
NumToStr fix with snprintf

### PR DESCRIPTION
**Pull request by schaumb**
'-0' is serialized as '0' + it can handle bigger numbers than ull max

### CODE REVIEW

**Comment by schaumb** [comment]
@veghdev Please update gcc to at least version 12 in test machine (newer is better). The emcc uses the gcc standard lib, and gcc10 is not C++20 compliant.

**Review by simzer — CHANGES_REQUESTED** [review]
This implementation increases the binary size with 107KB (27%). Most likely because it uses the locale.

**Comment by schaumb** [comment]
And this will not work, because switched off rtti.
-->

https://stackoverflow.com/questions/49939459/c-locale-without-rtti

**Comment by simzer** [comment]
Yep, but also, I think it would not prevent the compiler to link the needed infrastructure for locales.

**Review by simzer — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 - Fixed unintentional separate animation steps for specific dimension change
   animations.
 - Fixed JS exception mishandling as C++ exception when c++ segfault happens.
+- Fixed showing big doubles greater than UINT64_MAX.
```
[the remark this answers is not in the document]
Comment (schaumb) [reply CHANGELOG.md:20]: "Fixed showing big numbers greater than 18446744073709551615." is OK?

[the remark this answers is not in the document]
Comment (simzer) [reply CHANGELOG.md:20]: OK

**Review by simzer — DISMISSED** [review]

**Pull request merged**